### PR TITLE
Bug 1851186 - Add new design system color tokens

### PR DESCRIPTION
**Pull request by MozillaNoah**
Color figma and changelog are linked in the BZ ticket.

These tokens are needed for the shopping feature.

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1851186

### CODE REVIEW

**Comment by MozillaNoah** [comment]
Adding @cwzilla for transparency of our token addition and for some hex validation

File: fenix/app/src/main/java/org/mozilla/fenix/theme/FirefoxTheme.kt
Comment (gabrielluong) [inline fenix/app/src/main/java/org/mozilla/fenix/theme/FirefoxTheme.kt:177]: This is gonna sound somewhat crazy, but could I request that you reorder the parameters to follow the ordering in the [figma stickersheet](https://www.figma.com/file/pEyGeE4KV5ytYHeXMfLcEr/Mobile-Styles?type=design&node-id=1637-23964&mode=design&t=IjKu9IYFyEcnzFRv-0)?

My rationale is that I go up and down the stickersheet to just make sure everything is named correctly, has the right color, description, etc. Having the ordering thrown off when comparing our code with the figma would just make that comparison and check process that much harder.

Comment (gabrielluong) [reply fenix/app/src/main/java/org/mozilla/fenix/theme/FirefoxTheme.kt:177]: Specially, these layer variables need to go after `gradientEnd`

File: fenix/app/src/main/java/org/mozilla/fenix/theme/FirefoxTheme.kt
Comment (gabrielluong) [inline fenix/app/src/main/java/org/mozilla/fenix/theme/FirefoxTheme.kt:360]: Could we stick to the description used in the [stickersheet](https://www.figma.com/file/pEyGeE4KV5ytYHeXMfLcEr/Mobile-Styles?type=design&node-id=1637-23964&mode=design&t=IjKu9IYFyEcnzFRv-0)? 

I think we should leave it to the Design System team to describe what the variable means. If there is no description probably, then I would just avoid having as well.

File: fenix/app/src/main/java/org/mozilla/fenix/theme/FirefoxTheme.kt
Comment (gabrielluong) [inline fenix/app/src/main/java/org/mozilla/fenix/theme/FirefoxTheme.kt:374]: Same request as above about moving reordering these variables to match the ordering of the stickersheet.

File: fenix/app/src/main/java/org/mozilla/fenix/theme/FirefoxTheme.kt
Comment (gabrielluong) [inline fenix/app/src/main/java/org/mozilla/fenix/theme/FirefoxTheme.kt:100]: Let's add these tokens to `colors.xml` as well.

**Review by gabrielluong — CHANGES_REQUESTED** [review]
Changes requested.

File: fenix/app/src/main/res/values-night/colors.xml
Comment (gabrielluong) [inline fenix/app/src/main/res/values-night/colors.xml:45]:
```suggestion
    <color name="fx_mobile_action_color_warning" tools:ignore="UnusedResources">@color/photonYellow40A41</color>
```
Just aligning this with the token name that is listed in the stickersheet. We will need to do the same for the other button colors below.

Could you also correct the `fx_mobile_action_quarternary` above while we're at it?

File: fenix/app/src/main/java/org/mozilla/fenix/theme/FirefoxTheme.kt
Comment (gabrielluong) [inline fenix/app/src/main/java/org/mozilla/fenix/theme/FirefoxTheme.kt:692]: We should probably prefix these with `layer`. We could do this now or as a followup depending on your preference.

**Review by gabrielluong — APPROVED** [review]

**Comment by gabrielluong** [comment]
r+ some minor changes are needed, but this doesn't require another review.

**Comment by gabrielluong** [comment]
@MozillaNoah Could we also update the change log to indicate the changes have been updated in Fenix? https://docs.google.com/spreadsheets/d/1O_LHoSqGGeXor_e2y7dpsyldm9pNJrxEbp16mD8oMCg/edit#gid=1216550615

There is an "Updated in Fenix" in column used to indicate that the changes were updated in Fenix with the who and PR issue number it was fixed. This is used to keep track of the fact that we have implemented the changes that are listed in the changelog.

File: fenix/app/src/main/java/org/mozilla/fenix/theme/FirefoxTheme.kt
Comment (MozillaNoah) [reply fenix/app/src/main/java/org/mozilla/fenix/theme/FirefoxTheme.kt:692]: Let's do this as a follow-up since the changelog will likely get polluted from the usage changes.

I've opened https://bugzilla.mozilla.org/show_bug.cgi?id=1851654

**Pull request merged**
